frontend/src/components/Main/OptionsForm/OptionsForm.js

```diff
@@ -2,104 +2,79 @@ import React from "react";
 import PropTypes from "prop-types";
 
 import * as mc from "../../../constants/mainConstants";
+import SelectInput from "../../common/SelectInput";
 import {
     dichotomousBmrOptions,
     continuousBmrOptions,
     distTypeOptions,
 } from "../../../constants/optionsConstants";
+import IntegerInput from "../../common/IntegerInput";
 
 const OptionsForm = props => {
     return (
         <tr className="form-group">
             {props.modelType === mc.MODEL_CONTINUOUS ? (
                 <td>
-                    <select
-                        className="form-control"
-                        value={props.options.bmr_type}
-                        onChange={e =>
-                            props.saveOptions("bmr_type", parseInt(e.target.value), props.idx)
-                        }>
-                        {continuousBmrOptions.map(d => {
-                            return (
-                                <option key={d.value} value={d.value}>
-                                    {d.label}
-                                </option>
-                            );
+                    <SelectInput
+                        choices={continuousBmrOptions.map(option => {
+                            return {value: option.value, text: option.label};
                         })}
-                    </select>
+                        onChange={value =>
+                            props.saveOptions("bmr_type", parseInt(value), props.idx)
+                        }
+                        value={props.options.bmr_type}
+                    />
                 </td>
             ) : null}
             {props.modelType == mc.MODEL_DICHOTOMOUS ? (
                 <td>
-                    <select
-                        className="form-control"
-                        value={props.options.bmr_type}
-                        onChange={e =>
-                            props.saveOptions("bmr_type", parseInt(e.target.value), props.idx)
-                        }>
-                        {dichotomousBmrOptions.map(d => {
-                            return (
-                                <option key={d.value} value={d.value}>
-                                    {d.label}
-                                </option>
-                            );
+                    <SelectInput
+                        choices={dichotomousBmrOptions.map(option => {
+                            return {value: option.value, text: option.label};
                         })}
-                    </select>
+                        onChange={value =>
+                            props.saveOptions("bmr_type", parseInt(value), props.idx)
+                        }
+                        value={props.options.bmr_type}
+                    />
                 </td>
             ) : null}
 
             <td>
-                <input
-                    type="number"
-                    className="form-control text-center"
+                <IntegerInput
```

Comment:
you should use a FloatInput here, not an IntegerInput

Reply:
used FloatInput, but i made changes to the FloatInput.js file. we can discuss afterwards.